builder/main.py

```
@@ -297,6 +557,419 @@ def check_lib_archive_exists():
+
+    # Read sector size from FAT boot sector
+    sector_size = int.from_bytes(fs_data[0x0B:0x0D], byteorder='little')
+
+    if sector_size not in [512, 1024, 2048, 4096]:
+        print(f"Error: Invalid sector size {sector_size}. Must be 512, 1024, 2048, or 4096")
+        print("The filesystem may be corrupted or not a valid FAT filesystem")
+        return 1
+
+    print(f"  Detected sector size: {sector_size} bytes")
+
+    from fatfs import RamDisk, create_extended_partition
```

Ruff (0.14.10)

Redefinition of unused `create_extended_partition` from line 27

Remove definition: `create_extended_partition`

(F811)